src/search_algo.hpp

```diff
@@ -931,35 +927,34 @@ template <typename TLocalHolder>
 inline int
 iterateMatchesFullSimd(TLocalHolder & lH)
 {
-    using namespace seqan;
-
     using TGlobalHolder = typename TLocalHolder::TGlobalHolder;
 //     using TMatch        = typename TGlobalHolder::TMatch;
 //     using TPos          = typename TMatch::TPos;
     using TBlastPos     = uint32_t; //TODO why can't this be == TPos
-    using TBlastMatch   = BlastMatch<
+    using TBlastMatch   = seqan::BlastMatch<
                            typename TLocalHolder::TAlignRow0,
                            typename TLocalHolder::TAlignRow1,
                            TBlastPos,
-                           decltype(lH.gH.qryIds[0]),
-                           decltype(lH.gH.indexFile.subjIds[0]),
+                           seqan3::value_type_t<typename TGlobalHolder::TQryIds>,// TODO: reference_t?
+                           seqan3::value_type_t<typename TGlobalHolder::TSubjIds>// TODO: reference_t?
                            >;
-    using TBlastRecord  = BlastRecord<TBlastMatch,
-                                      decltype(lH.gH.qryIds[0]),
-                                      std::vector<std::string>,
-                                      decltype(lH.gH.indexFile.taxNames[0]),
-                                      uint32_t>;
+
+    using TBlastRecord  = seqan::BlastRecord<TBlastMatch,
+                                             seqan3::value_type_t<typename TGlobalHolder::TQryIds>,
+                                             std::vector<std::string>,
+                                             std::string, //seqan3::value_type_t<typename TGlobalHolder::TTaxNames>,
```

Comment:
Why did this change?

Reply:
I have changed this now to use views everywhere. It required a few more changes but avoid useless copies.